bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/BuildFolderVariableResolver.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright 2024 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.launch.serial.ui;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.QualifiedName;
+import org.eclipse.core.variables.IDynamicVariable;
+import org.eclipse.core.variables.IDynamicVariableResolver;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.launchbar.core.ILaunchBarManager;
+
+import com.espressif.idf.core.IDFConstants;
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.IDFDynamicVariables;
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.core.util.StringUtil;
+import com.espressif.idf.launch.serial.ui.internal.Activator;
+
+/**
+ * Class to resolve the build folder introduced for the flashing arguments as the cmake -B build directory when added to
+ * cmake build arguments needs to be added to the flash arguments as well
+ *
+ * @author Ali Azam Rana
+ *
+ */
+public class BuildFolderVariableResolver implements IDynamicVariableResolver
+{
+
+	private static final ILaunchBarManager LAUNCH_BAR_MANAGER = Activator.getService(ILaunchBarManager.class);
+
+	@Override
+	public String resolveValue(IDynamicVariable variable, String argument) throws CoreException
+	{
+		IDFDynamicVariables idfVariable = IDFDynamicVariables.valueOf(variable.getName());
+		if (idfVariable != IDFDynamicVariables.BUILD_DIR)
+		{
+			throw new RuntimeException("Invalid Variable for Resolver"); //$NON-NLS-1$
+		}
```

Comment:
Do we need this part? I think it's guaranteed that we are getting the "BUILD_DIR" dynamic variable here since this is the only variable that was specified in the extension point

Reply:
@sigmaaa yes we dont but its just incase this is accidentally used with some variable other than this it will be useful to show the exception